pairing_api/examples/pairing.rs

```diff
@@ -0,0 +1,216 @@
+        domain::Topic,
+        rpc::{params::ResponseParamsSuccess, Params, Payload},
+    },
+    std::{sync::Arc, time::Duration},
+    structopt::StructOpt,
+    tokio::{
+        signal,
+        spawn,
+        sync::mpsc::{unbounded_channel, UnboundedReceiver, UnboundedSender},
+    },
+    wc_common::decode_and_decrypt_type0,
+};
+
+#[derive(StructOpt)]
+struct Args {
+    /// Specify WebSocket address.
+    #[structopt(short, long, default_value = "wss://relay.walletconnect.com")]
+    address: String,
+
+    /// Specify WalletConnect project ID.
+    #[structopt(short, long, default_value = "1979a8326eb123238e633655924f0a78")]
+    project_id: String,
+}
+
+struct Handler {
+    name: &'static str,
+    sender: UnboundedSender<PublishedMessage>,
+}
+
+fn create_conn_opts(address: &str, project_id: &str) -> ConnectionOptions {
+    let key = SigningKey::generate(&mut rand::thread_rng());
+
+    let auth = AuthToken::new("http://127.0.0.1:8000")
+        .aud(address)
+        .ttl(Duration::from_secs(60 * 60))
+        .as_jwt(&key)
+        .unwrap();
+
+    ConnectionOptions::new(project_id, auth).with_address(address)
+}
+
+impl Handler {
+    fn new(name: &'static str, sender: UnboundedSender<PublishedMessage>) -> Self {
+        Self { name, sender }
+    }
+}
+
+impl ConnectionHandler for Handler {
+    fn connected(&mut self) {
+        println!("[{}] connection open", self.name);
+    }
+
+    fn disconnected(&mut self, frame: Option<CloseFrame<'static>>) {
+        println!("[{}] connection closed: frame={frame:?}", self.name);
+    }
+
+    fn message_received(&mut self, message: PublishedMessage) {
+        println!(
+            "[{}] inbound message: topic={} message={}",
+            self.name, message.topic, message.message
+        );
+
+        if let Err(err) = self.sender.send(message.clone()) {
+            println!("error {err:?} while sending {message:?}");
+        };
+    }
+
+    fn inbound_error(&mut self, error: ClientError) {
+        println!("[{}] inbound error: {error}", self.name);
+    }
+
+    fn outbound_error(&mut self, error: ClientError) {
+        println!("[{}] outbound error: {error}", self.name);
+    }
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let args = Args::from_args();
+    let (sender, receiver) = unbounded_channel();
+    let client1 = Arc::new(Client::new(Handler::new("client1", sender)));
+    client1
+        .connect(&create_conn_opts(&args.address, &args.project_id))
+        .await?;
+
+    let pairing_client = Arc::new(PairingClient::new());
+    // Create Pairing.
+    // let topic = create_pairing(&pairing_client).await;
+    // Pair
+    let topic = pair_from_uri(&pairing_client, &client1).await;
+    // Subscribe to the pairing topic
+    println!("\nSubscribing to topic: {}", topic);
+    client1
+        .subscribe(topic.clone())
+        .await
+        .map_err(PairingClientError::SubscriptionError)?;
```

Comment:
I guess here it's also possible to use #[from] for SubscriptionError.
Maybe there are also other map_err calls could be eliminated